test/spec/dataviews/dataview-model-base.spec.js

```diff
@@ -536,4 +568,51 @@ describe('dataviews/dataview-model-base', function () {
       expect(dataview.hasSameSourceAsLayer()).toBe(false);
     });
   });
+
+  describe('when analysis changes status', function () {
+    beforeEach(function () {
+      this.a0.isLoading = jasmine.createSpy('a0.isLoading');
+      this.a0.isFailed = jasmine.createSpy('a0.isFailed');
+      this.model.on({
+        loading: this.loadingSpy = jasmine.createSpy('loading'),
+        error: this.errorSpy = jasmine.createSpy('failed')
+      });
+    });
+
+    sharedTestsForAnalysisEvents();
+
+    describe('when changed source', function () {
+      beforeEach(function () {
+        this.model.set('source', {id: 'a0'});
```

Comment:
Considering the source is an _object_ and not a value we're technically creating a new object here, so the equality check would "fail" and it would set this as a "new' value and thus trigger an event.